packages/server/graphql/intranetSchema/mutations/checkRethinkPgEquality.ts

```diff
@@ -0,0 +1,74 @@
+import {GraphQLBoolean, GraphQLInt, GraphQLNonNull, GraphQLString} from 'graphql'
+import {requireSU} from '../../../utils/authorization'
+import fs from 'fs'
+import path from 'path'
+import checkUserEq from '../../../postgres/utils/checkUserEq'
+import checkTeamEq from '../../../postgres/utils/checkTeamEq'
+
+const tableResolvers = {
+  User: checkUserEq,
+  Team: checkTeamEq
+} as {
+  [key: string]: (
+    pageSize?: number,
+    startPage?: number,
+    slice?: boolean
+  ) => Promise<{[key: string]: any}>
+}
+
+const checkEqAndWriteOutput = async (
+  tableName: string,
+  fileLocation: string,
+  maxErrors: number = 10
+): Promise<void> => {
+  const errors = await tableResolvers[tableName](maxErrors)
+  await fs.promises.writeFile(fileLocation, JSON.stringify(errors))
+}
+
+const checkRethinkPgEquality = {
+  type: GraphQLNonNull(GraphQLString),
+  description: 'check equality of a table between rethinkdb and postgres',
+  args: {
+    tableName: {
+      type: GraphQLNonNull(GraphQLString),
+      description: 'The table name to be compared'
+    },
+    maxErrors: {
+      type: GraphQLInt,
+      default: 10,
+      description: 'How many errors should be returned'
+    },
+    writeToFile: {
+      type: GraphQLBoolean,
+      default: false,
+      description: 'Whether the output should be written to file'
+    }
+  },
+  resolve: async (_source, {tableName, maxErrors, writeToFile}, {authToken}) => {
+    // AUTH
+    requireSU(authToken)
+
+    // VALIDATION
+    if (!tableResolvers.hasOwnProperty(tableName)) {
+      return `That table name either doesn't exist or hasn't yet been implemented.`
+    }
+
+    // RESOLUTION
+    if (!writeToFile) {
+      const errors = await tableResolvers[tableName](maxErrors)
+      return JSON.stringify(errors)
+    }
+    const fileName = `${tableName}-${new Date()}`
+    const fileLocation = path.join(
+      process.cwd(),
+      'packages/server/graphql/intranetSchema/mutations',
+      '__output__',
+      fileName
+    )
+    await fs.promises.mkdir(path.dirname(fileLocation), {recursive: true})
+    checkEqAndWriteOutput(tableName, fileLocation, maxErrors)
```

Comment:
yes it's possible this fn while check the entire table and it will almost certainly time out. in that case, we have the option of setting `writeToFile=true` and the returned graphql msg will tell the caller where to find the file.